python/lsst/ts/cbp/csc.py

```diff
@@ -83,14 +82,32 @@ async def do_move(self, data):
         """
         self.log.debug("Begin move")
         self.assert_enabled("move")
-        await asyncio.join(
-            [
-                self.model.move_elevation(data.elevation),
-                self.model.move_azimuth(data.azimuth),
-            ]
+        await asyncio.gather(
+            self.model.move_elevation(data.elevation),
+            self.model.move_azimuth(data.azimuth),
+        )
+        self.evt_target.set_put(
+            azimuth=self.model.azimuth_target,
+            elevation=self.model.altitude_target,
+            focus=self.model.focus_target,
+            mask=self.model.mask_target,
+            mask_rotation=self.model.mask_rotation_target,
+        )
+        self.evt_inPosition.set_put(
+            azimuth=self.model.encoder_motion.azimuth,
+            elevation=self.model.encoder_motion.elevation,
+            focus=self.model.encoder_motion.focus,
+            mask=self.model.encoder_motion.mask_select,
+            mask_rotation=self.model.encoder_motion.mask_rotate,
+        )
+        await asyncio.wait_for(self.in_position(), 20)
+        self.evt_inPosition.set_put(
+            azimuth=self.model.encoder_motion.azimuth,
+            elevation=self.model.encoder_motion.elevation,
+            focus=self.model.encoder_motion.focus,
+            mask=self.model.encoder_motion.mask_select,
+            mask_rotation=self.model.encoder_motion.mask_rotate,
```

Comment:
I very strongly suggest centralizing your event-putting code and improving the robustness of the state reporting, to make sure the model state and reported state always match. Here are two techniques I have used that work:

* Pass the CSC to the model and have it use the CSC's event topics to store the state. That avoids duplication and simplifies state reporting. Instead of the CSC having to read all state from the model every time any state changes , the state is automatically reported when it changes. Concretely, in the model this means calling `self.csc.evt_foo.set_put(...)` to set state and `self.csc.evt_foo.data` to read state.

This is simple and robust, and minimizes duplication. You can use accessors to simplify getting and setting state, which adds some boilerplate code but makes the code more readable.

* Make the event reporting a separate method of your CSC and calling it *from the model* as a callback function, whenever the model updates any state.

This requires duplicating state between the model and the CSC, and requires that the CSC read all state any time any state changes.